src/liballoc/rc.rs

```diff
@@ -339,34 +358,49 @@ impl<T: ?Sized> Rc<T> {
 }
 
 impl<T: Clone> Rc<T> {
-    /// Make a mutable reference from the given `Rc<T>`.
+    #[inline]
+    #[unstable(feature = "rc_unique", reason = "renamed to Rc::make_mut", issue = "27718")]
+    #[deprecated(since = "1.4.0", reason = "renamed to Rc::make_mut")]
+    pub fn make_unique(&mut self) -> &mut T {
+        Rc::make_mut(self)
+    }
+
+    /// Make a mutable reference into the given `Rc<T>` by cloning the inner
+    /// data if the `Rc<T>` doesn't have one strong reference and no weak
+    /// references.
     ///
-    /// This is also referred to as a copy-on-write operation because the inner
-    /// data is cloned if the reference count is greater than one.
+    /// This is also referred to as a copy-on-write.
     ///
     /// # Examples
     ///
     /// ```
-    /// #![feature(rc_unique)]
-    ///
     /// use std::rc::Rc;
     ///
-    /// let mut five = Rc::new(5);
+    /// let mut data = Rc::new(5);
+    ///
+    /// *Rc::make_mut(&mut data) += 1;             // Won't clone anything
+    /// let mut other_data = data.clone(); // Won't clone inner data
+    /// *Rc::make_mut(&mut data) += 1;             // Clones inner data
+    /// *Rc::make_mut(&mut data) += 1;             // Won't clone anything
+    /// *Rc::make_mut(&mut other_data) *= 2;       // Won't clone anything
+    ///
+    /// // Note: data and other_data now point to different numbers
+    /// assert_eq!(*data, 8);
+    /// assert_eq!(*other_data, 12);
     ///
-    /// let mut_five = five.make_unique();
     /// ```
     #[inline]
-    #[unstable(feature = "rc_unique")]
-    pub fn make_unique(&mut self) -> &mut T {
-        if !Rc::is_unique(self) {
-            *self = Rc::new((**self).clone())
+    #[unstable(feature = "rc_unique", reason = "needs FCP", issue = "27718")]
+    pub fn make_mut(this: &mut Self) -> &mut T {
+        if !Rc::is_unique(this) {
+            *this = Rc::new((**this).clone())
```

Comment:
If there's only one strong pointer (this one) but many weak pointers then you can move ownership from the original `Rc` to a new one, thereby destroying the original one (decrementing its strong count to 0).

I believe `Arc` takes this strategy currently.